viewer/src/views/threejs/cad/RootSectorNode.ts

```diff
@@ -25,13 +25,17 @@ export class RootSectorNode extends SectorNode {
   private readonly materials: Materials;
 
   private readonly consumeSectorCache: MemoryRequestCache<string, ParsedSector, THREE.Group>;
+  private _nodeIdToTreeIndexMap: Map<number, number>;
+  private _treeIndexToNodeIdMap: Map<number, number>;
```

Comment:
Would an array be more efficient? I.e. Float64Array

Reply:
For the `treeIndexToNodeIdMap`, yes, but then you need to go from an intermittent `Map` in each sector and convert that to the Float64Array afterwards. I would strongly advise against storing `nodeIdToTreeIndexMap` as a Float64Array. It would be very sparse and potentially take up to 2^53 * 8 bytes to store. As soon as you insert a large node ID, it needs to allocate all the unused slots up to that ID.

Reply:
`nodeIdToThreeIndexMap` cannot be an array - totally agreed. However, `treeIndexToNodeIdMap` can and it would reduce allocations a lot as it can be allocated up front.

Reply:
Would you prefer that one is a map and the other a Float64Array? I was thinking it would be easier to work with if both had the same type, but we can make them different (this is the way it is in the old viewer).